homotopy-web/src/app/project.rs

```diff
@@ -50,6 +54,7 @@ impl Component for ProjectView {
                     {"Import"}
                 </label>
                 <input type="file" accept="application/msgpack,.hom,.json" class="visually-hidden" id="import" onchange={import}/>
+                <button onclick={actions}>{"Export Actions"}</button>
```

Comment:
I'm wondering if this should be in our debug menu?

Reply:
It most definitely should. However, I''m a webdev dumb dumb and couldn't figure it out, so any help would be appreciated on this.

Reply:
Thanks :)

Reply:
Haha I also said it should be in the debug panel. My only worry is that the debug is not available on the live preview which is slightly annoying.

Btw Chiara does the normal export contain the action history? If someone reports a bug and attaches the workspace export, we should be able to get the actions from that.

Reply:
It doesn't, but I agree it should. I think what it comes down to is that one of the data structures relating to the proof tree contains a timestamp of type Instant, which is not serialiser-friendly.

Reply:
Right, that's problematic because in practice it's unlikely people will construct a large diagram in one go without saving and re-loading (which would destroy the action history).

Reply:
Do you know why we are timestamping everything so aggressively? Several bits of code mention "determinism" but I have no idea what that has to do with anything.

Reply:
@doctorn ok so I gave another shot at moving the button and I can't figure out how to make the action dispatch happen. If anyone else wants to give it a shot, feel free to implement it, but I don't want to spend the whole day on it.

Reply:
> Do you know why we are timestamping everything so aggressively? Several bits of code mention "determinism" but I have no idea what that has to do with anything.

(CC @NickHu)

Reply:
> @doctorn ok so I gave another shot at moving the button and I can't figure out how to make the action dispatch happen. If anyone else wants to give it a shot, feel free to implement it, but I don't want to spend the whole day on it.

I can have a look after work?

Reply:
Sure, I'd really appreciate it. I will merge this in the meantime tho.